frontend/catalyst/utils/filesystem.py

```diff
@@ -53,10 +55,43 @@ def cleanup(self):
             return
         shutil.rmtree(str(self))
 
-    def __del__(self):
-        """Avoid warning by doing manual cleanup during deletion."""
-        if isinstance(self._impl, tempfile.TemporaryDirectory):
-            self._impl.cleanup()
+
+class TemporaryDirectorySilent(tempfile.TemporaryDirectory):
+    """Derived class from tempfile.TemporaryDirectory
+
+    This overrides the _cleanup method which would normally emit a warning
+    after removing the directory. This warning is unconditional and it is emitted
+    because it is not called explicitly.
+    """
+
+    @classmethod
+    def _cleanup(cls, name, warn_message, ignore_errors=False, delete=True):
+        """Ignore ResourceWarning during cleanup"""
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=ResourceWarning)
```

Comment:
Would it be simpler to just call the [cls._rmtree](https://github.com/python/cpython/blob/0691bd860d3de0c1feb1566d99315a0a582c62a0/Lib/tempfile.py#L935) function directly and skip the warning entirely?

Reply:
Since you agreed offline I'll just go ahead and resolve the conflict and then merge the PR :)